src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/report/ReportNode.kt

```diff
@@ -0,0 +1,83 @@
+package org.jqassistant.tooling.intellij.plugin.editor.report
+
+
+import org.jqassistant.schema.report.v2.ConstraintType
+import org.jqassistant.schema.report.v2.GroupType
+import org.jqassistant.schema.report.v2.ReferencableRuleType
+import org.jqassistant.schema.report.v2.RowType
+import java.util.*
+import javax.swing.tree.TreeNode
+
+open class ReportNode(private val parent: ReportNode?) : TreeNode {
+    private val children = mutableListOf<ReportNode>();
+
+    fun addChild(child: ReportNode) {
+        children.add(child)
+    }
+
+    override fun toString(): String {
+        return "child of: $parent"
+    }
+
+    override fun getChildAt(index: Int): TreeNode {
+        return children[index]
+    }
+
+    override fun getChildCount(): Int {
+        return children.size
+    }
+
+    override fun getParent(): ReportNode? {
+        return parent
+    }
+
+    override fun getIndex(p0: TreeNode?): Int {
+        return children.indexOf(p0)
+    }
+
+    override fun getAllowsChildren(): Boolean {
+        return true
+    }
+
+    override fun isLeaf(): Boolean {
+        return children.size == 0
+    }
+
+    override fun children(): Enumeration<out TreeNode> {
+        return Collections.enumeration(children)
+    }
+
+}
+
+open class ReferencableRuleTypeNode(val ref: ReferencableRuleType, parent: ReportNode?) : ReportNode(parent) {
+    constructor(ref: ReferencableRuleType) : this(ref, null) {
+    }
```

Comment:
Do we need a second constructor? Couldn't we just add `null` as default value to the standard one?